doc/PRACTICE_RECIPES.md

```diff
@@ -0,0 +1,78 @@
+# practice
+
+Practice recipes allow practicing skills and proficiencies by working on crafts that are "just for
+practice", and do not yield a finished result (although they may have some byproducts). They may be
+designed to focus on specific niche skills or proficiencies, which might be prohibitive to learn
+from the more productive (and time-consuming) crafting recipes requiring many proficiencies.
+
+The practice recipe JSON format is a form of the "recipe" type, with a few differences noted below.
+See [Recipes section of JSON_INFO](JSON_INFO.md#recipes) for more on recipe fields.
+
+
+## JSON fields
+
+Practice recipes have type "practice", and they cannot define a "result", since they are not
+intended to make a specific item. In most other ways, they are just like "recipe" definitions, are
+loaded into the main `recipe_dictionary` used for crafting, and appear in the crafting UI.
+
+They:
+
+- must define `id`, `name`, `description` and `practice_data`
+- must not define a `result` or `difficulty`
+- may define `byproducts`
+
+The `practice_data` field is required for practice recipes, and must be defined. It sets a minimum
```

Comment:
It might help to explicitly list the fields of `practice_data`: `min_difficulty`, `max_difficulty`, `skill_limit`
`min_difficulty` and `max_difficulty` are required
`skill_limit` is optional and defaults to the max skill level (10)

[Note to self: actually make `max_difficulty` required; it currently defaults to 9, which is almost never useful]

Reply:
Agreed, there is a lot to unpack for `practice_data` - I will embellish this section.